crates/librqbit/src/tracing_subscriber_config_utils.rs

```diff
@@ -81,7 +87,11 @@ pub fn init_logging(opts: InitLoggingOptions) -> anyhow::Result<InitLoggingResul
 
     let layered = tracing_subscriber::registry()
         // Stderr logging layer.
-        .with(fmt::layer().with_filter(stderr_filter))
+        .with(
+            fmt::layer()
+                .with_filter(stderr_filter)
+                .and_then(if let Format::Json = opts.format { Some(fmt::layer().json())} else { None } ),
```

Comment:
I'm not sure and_then works the way you think it does. You are probably logging to both layers - the JSON one and the regular one.

You probably need to make stderr and JSON layers mutually exclusive like [here](https://github.com/tokio-rs/tracing/issues/894):

```
let (json, plain) = if enable_json {
    (Some(tracing_subscriber::fmt::layer().json()), None)
} else {
    (None, Some(tracing_subscriber::fmt::layer()))
};
```